microhapulator/Snakefile

```diff
@@ -0,0 +1,354 @@
+            "MappingRate",
+            "MappedFullRefr",
+            "MappingRateFullRefr",
+            "Typed",
+            "TypingRate",
+            "InterlocChiSq",
+            "HetTstat",
+        )
+        data = {colname: [] for colname in columns}
+        for sample in sorted(config["samples"]):
+            print(f"[Compiling summary] Sample={sample}", file=sys.stderr)
+            totalreads, mergedreads, *_ = parse_flash_summary(f"analysis/{sample}/flash.log")
+            maptotal, mapped = parse_read_counts(f"analysis/{sample}/{sample}-mapped-reads.txt")
+            assert maptotal == mergedreads, (sample, maptotal, mergedreads)
+            frmaptotal, frmapped = parse_read_counts(
+        f"analysis/{sample}/fullrefr/{sample}-fullrefr-mapped-reads.txt"
+            )
+            assert (
+                frmaptotal == maptotal
+            ), f"Sample={sample} fullrefr map total={frmaptotal} map total={maptotal}"
+            typing = pd.read_csv(f"analysis/{sample}/{sample}-typing-rate.tsv", sep="\t")
+            num_typed_reads = typing.TypedReads.sum()
+            typing_total_reads = typing.TotalReads.sum()
+            # assert typing_total_reads == mapped, f"Sample={sample} type total={typing_total_reads} mapped={mapped}"
+            typing_rate = num_typed_reads / typing_total_reads
+            chisq = parse_balance_stat(f"analysis/{sample}/{sample}-interlocus-balance-chisq.txt")
+            tstat = parse_balance_stat(
+                f"analysis/{sample}/{sample}-heterozygote-balance-pttest.txt"
+            )
+            data["Sample"].append(sample)
+            data["TotalReads"].append(totalreads)
+            data["Merged"].append(mergedreads)
+            data["MergeRate"].append(mergedreads / totalreads)
+            data["Mapped"].append(mapped)
+            data["MappingRate"].append(mapped / maptotal)
+            data["MappedFullRefr"].append(frmapped)
+            data["MappingRateFullRefr"].append(frmapped / frmaptotal)
+            data["Typed"].append(num_typed_reads)
+            data["TypingRate"].append(typing_rate)
+            data["InterlocChiSq"].append(chisq)
+            data["HetTstat"].append(tstat)
+        df = pd.DataFrame(data)
+        df.to_csv(output.tsv, sep="\t", index=False, float_format="%.4f")
+
+
+rule fastqc:
+    input:
+        lambda wildcards: sorted([fq for fq in config["readfiles"] if wildcards.sample in fq]),
+    output:
+        r1="analysis/{sample}/fastqc/R1-fastqc.html",
+        r2="analysis/{sample}/fastqc/R2-fastqc.html",
+    threads: 2
+    run:
+        shell("fastqc --outdir analysis/{wildcards.sample}/fastqc/ --threads {threads} {input}")
+        outfiles = sorted(glob(f"analysis/{wildcards.sample}/fastqc/*.html"))
+        for end, outfile in enumerate(outfiles, 1):
+            outfile = Path(outfile)
+            linkfile = f"analysis/{wildcards.sample}/fastqc/R{end}-fastqc.html"
+            symlink(outfile.name, linkfile)
```

Comment:
Additional QC step not in any of the prototypes. Links to FastQC HTML reports are in the MicroHapulator report.